### PR TITLE
Add use current location functionality

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL:
https://dfedigital.atlassian.net/secure/RapidBoard.jspa?rapidView=21&projectKey=TEVA&modal=detail&selectedIssue=TEVA-204

This work adds the 'use your current location' functionality. Instead of adding an endpoint on our server we rely on [Postcodes.io](https://postcodes.io/) for getting the postcode from the coordinates we get from the browser.

Screenshot 2019-10-09 at 15 29 18
Screenshot 2019-10-09 at 15 29 30
Screenshot 2019-10-09 at 15 29 36
Screenshot 2019-10-09 at 15 29 43

### CODE REVIEW

**Comment by dushan-madetech** [comment]
I think it would be helpful to have some kind of output to the user if we get a 200 null response. Something like a message saying `"unable to determine location, please try again later"` as the current behaviour might cause users to think that the use my location link is broken

File: app/views/vacancies/_filters.html.haml [hunk 1]
```diff
@@ -41,8 +41,13 @@
...
     .govuk-form-group
-      = label_tag 'location', t('jobs.filters.location'), class: 'govuk-label'
-      = search_field_tag :location, location, class: 'govuk-input form-control form-control-block', placeholder: t('jobs.filters.location_hint')
+      .js-location-finder
+        = label_tag 'location', t('jobs.filters.location'), class: 'govuk-label'
+        .js-location-finder__input-container#js-location-finder__input-container
+          = search_field_tag :location, location, class: 'govuk-input form-control form-control-block js-location-finder__input', placeholder: t('jobs.filters.location_hint')
+        .js-location-finder__link
+          or
+          = link_to t('jobs.filters.use_current_location'), '#', id: 'current-location'
```
Comment (dushan-madetech) [inline app/views/vacancies/_filters.html.haml]: could you add these classes to the link to make the styling of the link better? `govuk-link govuk-link--no-visited-state`

File: documentation/adr/0001_get_postcode_from_coordinates.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Use postcodes.io to get postcode from coordinates
+Date: 10/10/2019
+
+## Status
+approved
+
+## Context
+We need to get a postcode from the coordiantes we get from the browser.
```
Comment (dushan-madetech) [inline documentation/adr/0001_get_postcode_from_coordinates.md]:
```suggestion
We need to get a postcode from the coordinates we get from the browser.
```

File: spec/features/job_seekers_can_use_current_location_link_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    page.execute_script <<-JS
+      #{file_fixture('jquery.mockjax.min.js').read}
+      $.mockjaxSettings.logging = 0;
+
+      $.mockjax({
+        url: "https://api.postcodes.io/postcodes",
+        status: #{response_type},
+        responseText: #{response_text}
+      });
+    JS
```
Comment (dushan-madetech) [inline spec/features/job_seekers_can_use_current_location_link_spec.rb]: Could we have mocked the postcodes.io calls using the VCR gem? (https://github.com/vcr/vcr) 

Mainly asking out of curiousity

File: app/assets/stylesheets/components/location_finder.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
+.js-location-finder {
+  .govuk-c-loader {
+    @extend .govuk-input;
+    @extend .form-control;
+    position: absolute;
+    top: 0;
+    left: 0;
+    // display: none;
```
Comment (adrianclay) [inline app/assets/stylesheets/components/location_finder.scss]: Commented code?!

**Pull request merged**